noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -194,11 +217,13 @@ impl LogEncryption for AES128 {
         // It is safe to call the `unsafe` function here, because we know the `shared_secret`
         // was derived using an AztecAddress (the recipient). See the block comment
         // at the start of this unsafe target function for more info.
-        let (sym_key, iv) = derive_aes_symmetric_key_and_iv_from_ecdh_shared_secret_using_poseidon2_unsafe(
+        let pairs = derive_aes_symmetric_key_and_iv_from_ecdh_shared_secret_using_poseidon2_unsafe::<2>(
             ciphertext_shared_secret,
         );
+        let (sym_key_1, iv_1) = pairs[0];
+        let (sym_key_2, iv_2) = pairs[1];
 
-        let ciphertext_bytes = aes128_encrypt(plaintext_bytes, iv, sym_key);
+        let ciphertext_bytes = aes128_encrypt(plaintext_bytes, iv_2, sym_key_2);
```

Comment:
Why 2 first and 1 second?

Reply:
My brain saw the header as 1 and the body as 2.

Reply:
`body_sym_key/iv`, `header_sym_key/iv`?